.travis.yml

```diff
@@ -44,6 +48,9 @@ addons:
       - dos2unix
       - lzip
       - rpm
+      - cppcheck
 
 script:
   - ./.travis.sh
+  # cppcheck
+  - cppcheck --quiet --error-exitcode=1 .
```

Comment:
Please use a new `MODE=cppcheck` instead and call `cppcheck` from `.travis.sh`, instead. This way, we run it once for every platform, not more.

Reply:
I added new mode, but I am not sure about calling them from .travis.sh file for all OS/platform. Could you please provide more details?

Reply:
Travis covers macOS and Linux for us so need to worry about Windows — that's AppVeyor.
Calling `cppcheck` from `.travis.sh` should work for both operating systems.
I was thinking of a new block similar to `elif [[ ${MODE} = cppcheck ]]; then`.
Does that answer the question?

Reply:
Yes, I was also thinking the same, but not sure how will it be called.

Added mode cppcheck below.

if [[ ${MODE} = distcheck ]]; then
    ./configure ${CONFIGURE_ARGS}
    make distcheck

    mkdir -p ~/rpmbuild/{BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
    ln -v -s "$PWD"/expat-*.tar.bz2 ~/rpmbuild/SOURCES/
    rpmbuild -ba expat.spec
elif [[ ${MODE} = cmake-oos ]]; then
    mkdir build
    cd build
    cmake ${CMAKE_ARGS} ..
    make all test
    make DESTDIR="${PWD}"/ROOT install
    find ROOT -printf "%P\n" | sort
elif [[ ${MODE} = cppcheck ]]; then
    cppcheck --quiet --error-exitcode=1 .
else
    ./qa.sh
fi

I am thinking to call it through .travis.yml file 

matrix:
  include:
    - os: linux
      env: MODE=cmake-oos CMAKE_ARGS=-DXML_ATTR_INFO=ON cppcheck
    - os: linux
      env: MODE=qa-sh QA_COMPILER=clang CC=clang CXX=clang++ LD=clang++ QA_SANITIZER=address CONFIGURE_ARGS=--enable-xml-attr-info cppcheck
    - os: linux
      env: MODE=qa-sh QA_COMPILER=clang CC=clang CXX=clang++ LD=clang++ QA_SANITIZER=memory cppcheck
    - os: linux
      env: MODE=qa-sh QA_COMPILER=clang CC=clang CXX=clang++ LD=clang++ QA_SANITIZER=undefined cppcheck
    - os: linux
      env: MODE=qa-sh QA_COMPILER=gcc   CC=gcc   CXX=g++     LD=ld      QA_PROCESSOR=gcov cppcheck

is this enough? or any other thoughts. Please let me know your thoughts.

Reply:
This part looks good:
```
elif [[ ${MODE} = cppcheck ]]; then
    cppcheck --quiet --error-exitcode=1 .
```
Please check in more detail how Travis matrix env works and make a new proposal.

Thanks!

Reply:
To be more concrete, a plain addition of ...
```
    - os: linux
      env: MODE=cppcheck
```
... to the matrix, would be what I'd expect — nothing more.

Reply:
Please check new pull request https://github.com/libexpat/libexpat/pull/290